File: laravel_bart/SchoolApp/app/Models/Employee.php
<?php

namespace App\Models;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;
use Tymon\JWTAuth\Contracts\JWTSubject;

class Employee extends Authenticatable implements JWTSubject
{
    use HasApiTokens, HasFactory, Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
      'f_name',
      'm_name',
      'l_name',
      'full_name',
      'address_id',
      'birthday',
      'phone',
      'email',
      'job_type',
      'edu_degree',
      'start_date',
      'end_date',
      'time_table',
      'image',
      'active',
      'password'
  ];
  protected $guard = 'employee';
  public function address()
  {
    return $this->belongsTo(Address::class);
  
  }
  
  public function note()
  {
    return $this->hasMany(Note::class);
  
  }
  
  public function absence_Let_emp()
  {
    return $this->hasMany(Absence_Let_emp::class);
  
  }
  
  public $with=['address']; 

    /**
     * The attributes that should be hidden for serialization.
     *
     * @var array<int, string>
     */
    protected $hidden = [
        'password',
        'remember_token',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'email_verified_at' => 'datetime',
    ];

        /**
     * Get the identifier that will be stored in the subject claim of the JWT.
     *
     * @return mixed
     */
    public function getJWTIdentifier()
    {
        return $this->getKey();
    }

    /**
     * Return a key value array, containing any custom claims to be added to the JWT.
     *
     * @return array
     */
    public function getJWTCustomClaims()
    {
        return [];
    }
}

File: flutter_bart/tr_app/lib/Screen/login/Models/employee.dart
// To parse this JSON data, do
//
//     final employee = employeeFromJson(jsonString);

import 'dart:convert';

Employee employeeFromJson(String str) => Employee.fromJson(json.decode(str));

String employeeToJson(Employee data) => json.encode(data.toJson());

class Employee {
  Employee(
      {required this.id,
      required this.fName,
      required this.mName,
      required this.lName,
      required this.active,
      required this.fullName,
      required this.addressId,
      required this.birthday,
      required this.phone,
      required this.email,
      required this.image,
      required this.jobType,
      required this.eduDegree,
      required this.startDate,
      required this.endDate});

  int id;
  String fName;
  String mName;
  String lName;
  int active;
  String fullName;
  int addressId;
  String birthday;
  String phone;
  String email;
  String image;
  String jobType;
  String eduDegree;
  String startDate;
  String endDate;

  factory Employee.fromJson(Map<String, dynamic> json) => Employee(
      id: json["id"],
      fName: json["f_name"],
      mName: json["m_name"],
      lName: json["l_name"],
      active: json["active"],
      fullName: json["full_name"],
      addressId: json["address_id"],
      birthday: json["birthday"],
      phone: json["phone"],
      email: json["email"],
      image: json["image"],
      jobType: json["job_type"],
      eduDegree: json["edu_degree"],
      startDate: json["start_date"],
      endDate: json["end_date"]);

  Map<String, dynamic> toJson() => {
        "id": id,
        "f_name": fName,
        "m_name": mName,
        "l_name": lName,
        "active": active,
        "full_name": fullName,
        "address_id": addressId,
        "birthday": birthday,
        "phone": phone,
        "email": email,
        "image": image,
        "job_type": jobType,
        "edu_degree": eduDegree,
        "start_date": startDate,
        "end_date": endDate
      };
}
